Automattic/jetpack
# Boost: Add Cloud CSS advanced recommendations

Fixes 39-gh-Automattic/boost-shield-api

#### Changes proposed in this Pull Request:

* Show Advanced recommendations for Cloud Critical CSS failed URLs.
* Refactor query arg addition for Cloud CSS generation URLs.
* Remove query args from the CLoud CSS URLs before storing the status.

#### Jetpack product discussion


NA

#### Does this pull request change what data or activity we track or use?


No

#### Testing instructions:






* Define constant `JETPACK_BOOST_CLOUD_CSS` as true to enable Cloud Critical CSS.
* Navigate to Jetpack boost dashboard: `wp-admin/admin.php?page=jetpack-boost`
* Enable `Automatically Optimize CSS Loading`
* Confirm that Critical CSS is generated.
* Simulate Critical CSS generation error for some URLs. Eg:
  ```
  if ( ! empty( $_GET['jb-generate-critical-css' ] ) && ( str_contains( $_SERVER['REQUEST_URI'], 'sample-page' ) ) ) {
	header("HTTP/1.0 500 Not Found");
	die();
  }
* Regenerate Cloud CSS by clicking the `regenerate` button
* Confirm that the appropriate error is displayed under the `Automatically Optimize CSS Loading` meta section with a link to the `advanced recommendations page`
* Navigate to the Advanced recommendation page and confirm that the recommendations are valid.
* Ensure that the showstopper error and the success status are not altered from this PR.
* Ensure that local Critical CSS works as expected.
